caffe2/share/contrib/observers/observer.cc

```diff
@@ -0,0 +1,142 @@
+#include "caffe2/share/contrib/observers/observer.h"
+#include "caffe2/share/contrib/observers/observer_config.h"
+
+#include "caffe2/core/common.h"
+#include "caffe2/core/init.h"
+#include "caffe2/core/operator.h"
+#include <random>
+
+namespace caffe2 {
+namespace {
+
+bool registerGlobalTimeNetObserverCreator(int* /*pargc*/, char*** /*pargv*/) {
+  SetGlobalNetObserverCreator([](NetBase* subject) {
+      return caffe2::make_unique<TimeNetObserver>(subject);
+    });
+    return true;
+}
+}
+
+REGISTER_CAFFE2_EARLY_INIT_FUNCTION(
+    registerGlobalTimeNetObserverCreator,
+    &registerGlobalTimeNetObserverCreator,
+    "Caffe2 net global observer creator");
+
+TimeNetObserver::TimeNetObserver(NetBase* subject_)
```

Comment:
Perhaps call this QPLObserver and template it on `<NetBase*>` and `<OperatorBase*>`?

Reply:
I think I need to change the name QPLObserver since it doesn't make much sense. I'm not sure about your comment on templating on <OperatorBase*>